### PR TITLE
Set onboarding status route

### PR DESCRIPTION
**Pull request by Wulingoh**
Added route documentation and updated merchant model

image

image

Test Plans:
Check the route and table are showed up as expected.

### CODE REVIEW

File: legacy/src/api/merchants/merchants.md [hunk 1]
```diff
@@ -29,24 +29,38 @@ which define the payment methods available for a Payment Request.
...
+| location               | {% dt Location %}                       | **EXPERIMENTAL** Physical Location of Merchant.                            |
+| onboardingStatusReason | String                                  | The reason associated with the [Onboarding Status](#onboarding-statuses). |
+
+<a name="onboarding-statuses">
+{% h4 Onboarding Statuses %}
+
+|    Status    |                                                                           Description                                                                            |                                Allowed Reasons                                 |
+| :----------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------- | :----------------------------------------------------------------------------- |
+| applied      | Identifies merchants that have registered via the Merchant portal with limited information. Requires Customer Support assistance to progress.                    |                                                                                |
```
Comment (Ray2126) [inline legacy/src/api/merchants/merchants.md:58]: Will someone looking at our docs know what Customer support is? Seems like something we don't need to mention publicly

Comment (Ray2126) [reply legacy/src/api/merchants/merchants.md:58]: Ditto for Merchant portal.

We could just say - `Identifies merchants that have registered to be activated for Centrapay payments`

**Review by Ray2126 — CHANGES_REQUESTED** [review]

File: legacy/src/api/merchants/merchants.md [hunk 2]
```diff
@@ -29,24 +29,38 @@ which define the payment methods available for a Payment Request.
...
-| settlementConfig | [Settlement Config](#settlement-config) | **EXPERIMENTAL** Merchant settlement config               |
-| location         | {% dt Location %}                       | **EXPERIMENTAL** Physical Location of Merchant            |
+|         Field          |                  Type                   |                                Description                                |
+| :--------------------- | :-------------------------------------- | :------------------------------------------------------------------------ |
+| test                   | Boolean                                 | **EXPERIMENTAL** Flag indicating merchant is for testing.                 |
+| settlementConfig       | [Settlement Config](#settlement-config) | **EXPERIMENTAL** Merchant settlement config.                               |
+| location               | {% dt Location %}                       | **EXPERIMENTAL** Physical Location of Merchant.                            |
+| onboardingStatusReason | String                                  | The reason associated with the [Onboarding Status](#onboarding-statuses). |
```
Comment (tom-blake) [inline legacy/src/api/merchants/merchants.md:51]: We should have descriptions for these values

**Review by tom-blake — CHANGES_REQUESTED** [review]

File: legacy/src/api/merchants/merchants.md (hunk 2 above)
Comment (AndyClifford) [reply legacy/src/api/merchants/merchants.md:51]: Do you mean a separate table with each possible `onboardingStatusReason` with a relevant description?

Comment (tom-blake) [reply legacy/src/api/merchants/merchants.md:51]: Yep - that's what I was thinking

Comment (AndyClifford) [reply legacy/src/api/merchants/merchants.md]: I've added a table with each reason and placeholder descriptions - open to modifications. I wasn't sure on a couple of the descriptions e.g. `centrapay-discontinued` - was there a notion doc somewhere with more details on each of these? I couldn't find it

Comment (AndyClifford) [reply legacy/src/api/merchants/merchants.md]: image

**Review by tom-blake — APPROVED** [review]

**Pull request merged**